nautobot/docs/plugins/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,285 @@
+        example_models = ExampleModel.objects.filter(site__isnull=False).select_related("site", "location")
+        for  example_model  in  example_models:
+            site_name = example_model.site.name
+            # Point the location field to the corresponding "Site" LocationType Location
+            # with the same name.
+            example_model.location = Location.objects.get(location_type=site_location_type, name=site_name)
+            ExampleModel.objects.bulk_update(example_models, ["location"], 1000)
+
+class Migration(migrations.Migration):
+    dependencies = [
+        # The dcim migration creates the Site Type and Region Type Locations that
+        # your data models are migrating to. It has to be run **before** this migration.
+        ("dcim", "0030_migrate_region_and_site_data_to_locations"),
+        ("example_plugin", "0007_add_location_field_to_example_model"),
+    ]
+    operations = [
+        migrations.RunPython(
+            # Execute the function
+            code=migrate_example_model_data_to_locations,
+            reverse_code=migrations.operations.special.RunPython.noop,
+        )
+    ]
+
+```
+
+## Remove Site/Region Related Fields from Migrated Data Models
+
+After you data migration is successful, we need to remove the `site`/`region` fields from your data model. You can do that by simply removing the `site`/`region` attributes from your model class:
+
+```python
+
+# models.py
+class  ExampleModel(OrganizationalModel):
+    # note that site field is gone
+    location = models.ForeignKey(to="dcim.Location", on_delete=models.PROTECT, blank=True, null=True)
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+    number = models.IntegerField(default=100, help_text="The number of this Example.")
+    csv_headers = ["name", "number"]
+
+    class  Meta:
+        ordering = ["name"]
+...
+
+```
+
+After removing the `site` attribute, make the migration file by running:
+
+```shell
+
+nautobot-server makemigrations [app_name] -n [migration_name]
+nautobot-server makemigrations example_plugin -n remove_site_field_from_example_model
+
+```
+
+The migration file might look like this:
+
+```python
+
+# Generated by Django 3.2.17 on 2023-02-22 17:09
+# 0009_remove_site_field_from_example_model.py
+
+from  django.db  import  migrations
+
+class  Migration(migrations.Migration):
+
+    dependencies = [
+        ('example_plugin', '0008_migrate_example_model_data_from_site_to_location'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='examplemodel',
+            name='site',
+        ),
+    ]
+
+```
+
+!!! important
+    Before you apply this migration, you have to add an `run_before` attribute in this migration file to make sure that you remove `site`/`region` fields before `Site` and `Region` models themselves are removed. **Without it, your migration files might be out of order and your app will not start**.
+
```

Comment:
This raises an important question - how does this work if we have these migrations (including the `run_before`) in an app **that we install for the first time AFTER Nautobot has already migrated to 2.0**?

Reply:
Hmmm, I don't see a way for apps that are developed based on Nautobot v1.x to be installed after Nautobot has already migrated to v2.0.
1. Either revert Nautobot to v1.x, install the app, and then migrate to v2.0 together. (I don't even think the reversion is possible with the number of migrations)
2. Install the apps before migrating Nautobot to v2.0. 
3. Develop new apps that are the same as the old ones but v2.0 compatible?